In [1]:
%matplotlib inline

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
# Import database
df = pd.read_csv('Oscars/database.csv')

In [3]:
# Inspect
df

Year  Ceremony                             Award  Winner  \
0     1927/1928         1                             Actor     NaN   
1     1927/1928         1                             Actor     1.0   
2     1927/1928         1                           Actress     NaN   
3     1927/1928         1                           Actress     1.0   
4     1927/1928         1                           Actress     NaN   
5     1927/1928         1                     Art Direction     NaN   
6     1927/1928         1                     Art Direction     1.0   
7     1927/1928         1                     Art Direction     NaN   
8     1927/1928         1                    Cinematography     NaN   
9     1927/1928         1                    Cinematography     1.0   
10    1927/1928         1                    Cinematography     1.0   
11    1927/1928         1        Directing (Comedy Picture)     1.0   
12    1927/1928         1        Directing (Comedy Picture)     NaN   
13    1927/1928         1      Directing (Dramatic Picture)     1.0   
14    1927/1928         1      Directing (Dramatic Picture)     NaN   
15    1927/1928         1      Directing (Dramatic Picture)     NaN   
16    1927/1928         1               Engineering Effects     NaN   
17    1927/1928         1               Engineering Effects     1.0   
18    1927/1928         1               Engineering Effects     NaN   
19    1927/1928         1               Outstanding Picture     NaN   
20    1927/1928         1               Outstanding Picture     NaN   
21    1927/1928         1               Outstanding Picture     1.0   
22    1927/1928         1       Unique and Artistic Picture     1.0   
23    1927/1928         1       Unique and Artistic Picture     NaN   
24    1927/1928         1       Unique and Artistic Picture     NaN   
25    1927/1928         1              Writing (Adaptation)     NaN   
26    1927/1928         1              Writing (Adaptation)     NaN   
27    1927/1928         1              Writing (Adaptation)     1.0   
28    1927/1928         1          Writing (Original Story)     NaN   
29    1927/1928         1          Writing (Original Story)     1.0   
...         ...       ...                               ...     ...   
9934       2015        88          Short Film (Live Action)     NaN   
9935       2015        88          Short Film (Live Action)     1.0   
9936       2015        88                     Sound Editing     1.0   
9937       2015        88                     Sound Editing     NaN   
9938       2015        88                     Sound Editing     NaN   
9939       2015        88                     Sound Editing     NaN   
9940       2015        88                     Sound Editing     NaN   
9941       2015        88                      Sound Mixing     NaN   
9942       2015        88                      Sound Mixing     1.0   
9943       2015        88                      Sound Mixing     NaN   
9944       2015        88                      Sound Mixing     NaN   
9945       2015        88                      Sound Mixing     NaN   
9946       2015        88                    Visual Effects     1.0   
9947       2015        88                    Visual Effects     NaN   
9948       2015        88                    Visual Effects     NaN   
9949       2015        88                    Visual Effects     NaN   
9950       2015        88                    Visual Effects     NaN   
9951       2015        88      Writing (Adapted Screenplay)     1.0   
9952       2015        88      Writing (Adapted Screenplay)     NaN   
9953       2015        88      Writing (Adapted Screenplay)     NaN   
9954       2015        88      Writing (Adapted Screenplay)     NaN   
9955       2015        88      Writing (Adapted Screenplay)     NaN   
9956       2015        88     Writing (Original Screenplay)     NaN   
9957       2015        88     Writing (Original Screenplay)     NaN   
9958       2015        88     Writi

In [4]:
# Drop unnecessary columns
df = df[['Award','Winner','Name','Film']]

In [5]:
df['Award'].unique()

array(['Actor', 'Actress', 'Art Direction', 'Cinematography',
       'Directing (Comedy Picture)', 'Directing (Dramatic Picture)',
       'Engineering Effects', 'Outstanding Picture',
       'Unique and Artistic Picture', 'Writing (Adaptation)',
       'Writing (Original Story)', 'Writing (Title Writing)',
       'Special Award', 'Directing', 'Writing', 'Outstanding Production',
       'Sound Recording', 'Short Subject (Cartoon)',
       'Short Subject (Comedy)', 'Short Subject (Novelty)',
       'Assistant Director', 'Film Editing', 'Music (Scoring)',
       'Music (Song)', 'Dance Direction', 'Writing (Screenplay)',
       'Actor in a Supporting Role', 'Actress in a Supporting Role',
       'Short Subject (Color)', 'Short Subject (One Reel)',
       'Short Subject (Two Reel)', 'Irving G. Thalberg Memorial Award',
       'Music (Original Score)', 'Cinematography (Black and White)',
       'Cinematography (Color)', 'Special Effects',
       'Art Direction (Black and White)', 'Art Direct

In [6]:
# Replace all the variant names of Best Picture into Best Picture
bp_variants = {'Outstanding Picture', 'Outstanding Production', 'Outstanding Motion Picture', 'Best Motion Picture', 'Best Picture'}

for index, row in df.iterrows():
    if(row['Award'] in bp_variants):
        df.loc[index,'Award'] = 'Best Picture'

C:\Users\dell\Anaconda3\lib\site-packages\pandas\core\indexing.py:517: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[item] = s


In [7]:
# Only keep best picture awards
df = df[df['Award']=='Best Picture']

In [8]:
df.shape

(528, 4)

In [9]:
print((df[df['Winner'] == 1]).to_string())

             Award  Winner                                              Name                                               Film
21    Best Picture     1.0                            Paramount Famous Lasky                                              Wings
64    Best Picture     1.0                               Metro-Goldwyn-Mayer                                The Broadway Melody
100   Best Picture     1.0                   All Quiet on the Western Front                                           Universal
140   Best Picture     1.0                                         Cimarron                                           RKO Radio
178   Best Picture     1.0                                      Grand Hotel                                 Metro-Goldwyn-Mayer
237   Best Picture     1.0                                        Cavalcade                                                 Fox
301   Best Picture     1.0                            It Happened One Night                             

We noticed that there are 89 pictures instead of the expected 88. We went through them by hand and found that M\*A\*S\*H was put under winning when it should not have. Also there are movies in which the producer is mistakenly put as film name.

In [10]:
# M*A*S*H was not a winner
df.loc[4852,'Winner']=np.nan
# Change to correct film names
df.loc[21,'Name']="Wings"
df.loc[64,'Name']="The Broadway Melody"

In [11]:
print(df.to_string())

             Award  Winner                                               Name                                               Film
19    Best Picture     NaN                                  The Caddo Company                                         The Racket
20    Best Picture     NaN                                                Fox                                         7th Heaven
21    Best Picture     1.0                                              Wings                                              Wings
62    Best Picture     NaN                                Feature Productions                                              Alibi
63    Best Picture     NaN                                                Fox                                     In Old Arizona
64    Best Picture     1.0                                The Broadway Melody                                The Broadway Melody
65    Best Picture     NaN                                Metro-Goldwyn-Mayer                    

Several of the nominations had wrong names too...

In [12]:
df.loc[19,'Name']="The Racket"
df.loc[20,'Name']="7th Heaven"
df.loc[62,'Name']="Alibi"
df.loc[63,'Name']="In Old Arizona"
df.loc[65,'Name']="Hollywood Revue"
df.loc[66,'Name']="The Patriot"

In [13]:
# Now we can drop the name row to only have film name
df = df[['Name','Winner']]

In [20]:
df['Name'].unique().size

523

In [22]:
df.shape

(528, 2)

waIT, there are more entries than unique movie names?

In [23]:
df['Name'].value_counts()

Cleopatra                                           2
Moulin Rouge                                        2
Mutiny on the Bounty                                2
Heaven Can Wait                                     2
Romeo and Juliet                                    2
Decision before Dawn                                1
Butch Cassidy and the Sundance Kid                  1
The Awful Truth                                     1
The Tree of Life                                    1
The Little Foxes                                    1
The Thin Red Line                                   1
The Full Monty                                      1
Titanic                                             1
Watch on the Rhine                                  1
The Fighter                                         1
Five Star Final                                     1
The Big House                                       1
Network                                             1
Cries and Whispers          

This is because there are movies of the same name that got nominated/won for Best Picture in different years! But that's OK because it's just 5 entires out of 528!!

In [43]:
df.set_index('Name',inplace=True)